File: 2021/day1.rb
readings = Array.new

File.open('test.txt', 'r') do |f|
    f.each_line do |line|
        readings.push(line.to_i)
    end
end

#
# Part 1: Single measurement
#
prev = 0
count = -1
readings.each do |reading|
    if reading > prev
        count += 1
    end
    prev = reading
end

puts count

#
# Part 2: Sliding window of three
#
prev = 0
count = -1
readings.each_with_index do |reading, idx|
    measurement = readings[idx, 3].sum()
    if measurement > prev
        count += 1
    end
    prev = measurement
end

puts count

File: 2021/Day1/day1.py
SLIDING_WINDOW_OFFSET = 3


if __name__ == '__main__':
    with open('test.txt') as f:
        readings = [int(reading.strip()) for reading in f.readlines()]

    #
    # Part 1: Single measurement
    #
    prev = 0
    count = -1
    for reading in readings:
        if reading > prev:
            count += 1
        prev = reading
    print(f'Part 1: {count}')


    #
    # Part 2: Sliding window of three
    #
    prev = 0
    count = -1
    for idx, _ in enumerate(readings):
        measurement = sum(readings[idx:idx+SLIDING_WINDOW_OFFSET])
        if measurement > prev:
            count += 1
        prev = measurement
    print(f'Part 2: {count}')
